### ISSUE TITLE
dropout(inplace=True) raises error : one of the variables needed for gradient computation has been modified by an inplace operation:Tensor [], which is output 0 of ReluBackward0, is at version 2; expected version 1 instead.

### ISSUE THREAD

**Issue by GitZinger** [pytorch/pytorch#72992]
### 🐛 Describe the bug

Invoking the nn.Dropout(..., inplace=True) will make the training crash 

one of the variables needed for gradient computation has been modified by an inplace operation:Tensor [], 
which is output 0 of ReluBackward0, is at version 2; expected version 1 instead.

```
self.main = nn.Sequential(         
nn.Conv2d(...., bias=False),
nn.LeakyReLU(0.2, inplace=True),
nn.Dropout(0.25, inplace=True), 
nn.Conv2d(....., bias=False),

)
```

### Versions

PyTorch version: 1.10.2
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Microsoft Windows 10 Enterprise
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect
Libc version: N/A

Python version: 3.7.6 (default, Jan  8 2020, 20:23:39) [MSC v.1916 64 bit (AMD64)] (64-bit runtime)
Python platform: Windows-10-10.0.19041-SP0
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: NVIDIA GeForce GTX 1070
Nvidia driver version: 511.79
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A

Versions of relevant libraries:
[pip3] numpy==1.18.1
[pip3] numpydoc==0.9.2
[pip3] torch==1.10.2
[pip3] torchaudio==0.10.2
[pip3] torchinfo==1.6.3
[pip3] torchvision==0.11.3
[conda] blas                      1.0                         mkl
[conda] cudatoolkit               11.3.1               h59b6b97_2
[conda] mkl                       2020.0                      166
[conda] mkl-service               2.3.0            py37hb782905_0
[conda] mkl_fft                   1.0.15           py37h14836fe_0
[conda] mkl_random                1.1.0            py37h675688f_0
[conda] numpy                     1.18.1           py37h93ca92e_0
[conda] numpy-base                1.18.1           py37hc3f5095_1
[conda] numpydoc                  0.9.2                      py_0
[conda] pytorch                   1.10.2          py3.7_cuda11.3_cudnn8_0    pytorch
[conda] pytorch-mutex             1.0                        cuda    pytorch
[conda] torchaudio                0.10.2               py37_cu113    pytorch
[conda] torchinfo                 1.6.3              pyhd8ed1ab_0    conda-forge
[conda] torchvision               0.11.3               py37_cu113    pytorch

**Comment by gchanan** [comment]
This is expected, you may ask for help on discuss.pytorch.org.
